trackintel/preprocessing/triplegs.py

```diff
@@ -292,6 +255,58 @@ def _seperate_ids(row):
     return sp, tpls, trips
 
 
+def _concat_staypoints_triplegs(staypoints, triplegs, add_geometry):
+    """Concatenate staypoints and triplegs to sp_tpls with new columns ["type", "is_activity", "sp_tpls_id"].
+
+    Parameters
+    ----------
+    staypoints : GeoDataFrame (as trackintel staypoints)
+    triplegs : GeoDataFrame (as trackintel triplegs)
+    add_geometry : bool
+
+    Returns
+    -------
+    sp_tpls : GeoDataFrame
+        "type": either "staypoint" or "tripleg"
+        "is_activity": True for staypoints that are activity staypoints
+        "sp_tpls_id": id of the corresponding staypoint or tripleg
+
+    Raises
+    ------
+    AttributeError
+        Raised if staypoints don't have column "is_activity"
+    """
+    if "is_activity" not in staypoints:
+        raise AttributeError("staypoints need the column 'is_activity' to be able to generate trips")
+    # Copy the input because we add temporary column "type"
+    tpls = triplegs.copy()
+    sp = staypoints.copy()
+
+    # write warnings for columns that we replace
+    if "trip_id" in tpls:
+        warnings.warn(f"Override column 'trip_id' in copy of triplegs.")
+
+    intersection = sp.columns.intersection(["trip_id", "prev_trip_id", "next_trip_id"])
+    if len(intersection):
+        warnings.warn(f"Override column(s) {intersection} in copy of staypoints.")
+
+    tpls["is_activity"] = False  # in case "is_activity" is already a column of tpls
+    tpls["type"] = "tripleg"
+    sp["type"] = "staypoint"
+
+    # create table with relevant information from triplegs and staypoints.
+    sp_cols = ["started_at", "finished_at", "user_id", "type", "is_activity"]
+    tpls_cols = ["started_at", "finished_at", "user_id", "type"]
+    sp_tpls = pd.concat([sp[sp_cols], tpls[tpls_cols]])
+    sp_tpls["is_activity"].fillna(False, inplace=True)
+    sp_tpls["sp_tpls_id"] = sp_tpls.index  # store id for later reassignment
+    if add_geometry:
+        sp_tpls["geom"] = pd.concat([sp.geometry, tpls.geometry])
```

Comment:
Do you mean instead of doing the concatinating in two steps we could do it in one like below?
```python
sp_cols = ["started_at", "finished_at", "user_id", "type", "is_activity"]
tpls_cols = ["started_at", "finished_at", "user_id", "type"]
if add_geometry:
    sp_cols.append(sp.geometry.name)
    tpls_cols.append(tpls.geometry.name)
sp_tpls = pd.concat([sp[sp_cols], tpls[tpls_cols]])
```

I thought about that but I think the efficiency gains are not that big. But what do you think?

Reply:
Hm... yeah you are right. It probably doesn't matter. I'll merge it then.